modules/currency.js

```diff
@@ -0,0 +1,220 @@
+
+import bidfactory from 'src/bidfactory';
+import { STATUS } from 'src/constants';
+import { ajax } from 'src/ajax';
+import * as utils from 'src/utils';
+import bidmanager from 'src/bidmanager';
+
+$$PREBID_GLOBAL$$.currency = setConfig;
+
+const DEFAULT_CURRENCY_RATE_URL = 'http://currency.prebid.org/latest.json';
+const CURRENCY_RATE_PRECISION = 4;
+
+var bidResponseQueue = [];
+var conversionCache = {};
+var currencyRatesLoaded = false;
+var adServerCurrency = 'USD';
+
+// Used as reference to the original bidmanager.addBidResponse
+var originalBidResponse;
+
+export var currencySupportEnabled = false;
+export var currencyRates = {};
+var currencyOverrides = {};
+
+export function setConfig(config) {
+  var url = DEFAULT_CURRENCY_RATE_URL;
+  if (typeof config.adServerCurrency === 'string') {
+    utils.logInfo('enabling currency support', arguments);
+
+    adServerCurrency = config.adServerCurrency;
+    if (config.conversionRateFile) {
+      utils.logInfo('currency using override conversionRateFile:', config.conversionRateFile);
+      url = config.conversionRateFile;
+    }
+    initCurrency(url);
+  } else {
+    // currency support is disabled, setting defaults
+    utils.logInfo('disabling currency support');
+    resetCurrency();
+  }
+  if (typeof config.currencyOverrides === 'object') {
+    currencyOverrides = config.currencyOverrides;
+  }
+}
+
+function initCurrency(url) {
+  conversionCache = {};
+  currencySupportEnabled = true;
+
+  if (!originalBidResponse) {
+    utils.logInfo('Installing addBidResponse decorator for currency module', arguments);
+
+    originalBidResponse = bidmanager.addBidResponse;
+    bidmanager.addBidResponse = addBidResponseDecorator(bidmanager.addBidResponse);
```

Comment:
Thanks for the ideas @muuki88, but we've decided to move forward with the design as noted in https://github.com/prebid/Prebid.js/issues/1089, modified slightly to use the new setConfig() approach. We think this has the advantage of preserving the bidCpmAdjustment for bidder-specific gross-vs-net conversions, separating currency which really isn't bidder-specific.

As for debugging, @snapwich has already implemented what we think is a pretty solid approach -- the bidResponse gets two new attributes: originalCpm and originalCurrency. Then cpm and currency are overwritten with the converted values. So you'll be able to access the before/after bidResponse attributes from the console.